### ISSUE TITLE
BufferedMetricsLogger can't be used as a type

### ISSUE THREAD

**Issue by jfirebaugh** [dbader/node-datadog-metrics#119]
The following should work ([TypeScript playground link](https://www.typescriptlang.org/play/?noLib=true&jsx=0&ts=5.4.2#code/JYWwDg9gTgLgBAbzgIQK4DN0FMpYCYCyWMUwAxgM4AyEA5rTnAL5zpQQhwDkeAhjLzx0AtCGKlKXAFBT0qAHZkYwCPLioKWGvRwAKADZ0GUAFwoM2XIXHlqRnAEpEUuK8M6oAOlq9UDXVwA1lgAnlwANHAADA5STDIaWvZQuvJYAO7mmDj4RCS22sa6DrFAA)):

```ts
import { BufferedMetricsLogger } from 'datadog-metrics'

function useLogger(logger: BufferedMetricsLogger) {
   logger.gauge('key', 0)
}

useLogger(new BufferedMetricsLogger())
```

However, it produces the following error:

> 'BufferedMetricsLogger' refers to a value, but is being used as a type here. Did you mean 'typeof BufferedMetricsLogger'?

(The suggestion the error message provides is not what I meant -- it generates [different errors](https://www.typescriptlang.org/play/?noLib=true&jsx=0&ts=5.4.2#code/JYWwDg9gTgLgBAbzgIQK4DN0FMpYCYCyWMUwAxgM4AyEA5rTnAL5zpQQhwDkeAhjLzx0AtCGKlKXAFBT0qAHZkYwCPLioKWGvRwAKADZ0GUAFxwYATzBYI6FBmy5C48tSM4AlIilxfhnVAAdLS8qAy6XADWWBZcADRwAAweUkwyGlruULryWADu9pg4+EQkrtrGuh4pQA).)

**Comment by jfirebaugh** [comment]
The correct way to do this is given [here](https://github.com/Microsoft/TypeScript/issues/10058#issuecomment-236458961):

```ts
export declare const BufferedMetricsLogger = loggers.BufferedMetricsLogger;
export declare type BufferedMetricsLogger = loggers.BufferedMetricsLogger;
```

**Comment by jfirebaugh** [comment]
[Workaround until this is fixed](https://www.typescriptlang.org/play/?noLib=true&jsx=0&ts=5.4.2#code/JYWwDg9gTgLgBAbzgIQK4DN0FMpYCYCyWMUwAxgM4AyEA5rTnAL5zpQQhwDkeAhjLzx0AtCGKlKXAFChIsRCgzZchceWp0GUOLwqLMOfERLqa9HABUAnmCzNW7Tj36CRYk5QD0eYBRieAG2AAI0DNHAppKXRUADsyGGAIWLhUCiwzLQAKAPCoAC59ZSM1SkzLGywASkQpOHrc8ygAOlpeVAYsrgBrLCsuABo4AAYqqSYpKTSMvKzYrAB3IsNVDw0mrKqxoA)

**Comment by Mr0grog** [comment]
Hey @jfirebaugh! 👋 

Thanks for filing this! Guess I should have included a test of the types when I added support for generating them here. 😓 All the typings here are generated from JSDoc, so I need to figure out the right incantation to get it to do this. Will see what I can figure that out ASAP.

**Comment by jfirebaugh** [comment]
Hi Rob! Good to see you around. 😄

**Comment by Mr0grog** [comment]
@jfirebaugh I think I’ve got a working solution over in #120; are you able to test that that works for you, or do you need a pre-release build on NPM?

**Comment by jfirebaugh** [comment]
I think I would need a pre-release build. I tried putting `"datadog-metrics": "dbader/node-datadog-metrics#119-if-you-want-types-you-probably-want-actual-types"` in my package.json but the resulting installed package does not have declaration files. Nor does it have a tsconfig.json, so running `cd node_modules/datadog-metrics && npm run build-types` doesn't work either.

**Comment by Mr0grog** [comment]
@jfirebaugh OK, you should be able to test with `0.11.4-a1`. (It’s clearly been too long since I published a pre-release on NPM, because I totally botched this one, hence version 0.11.4 instead of 0.11.3. 😩)

**Comment by jfirebaugh** [comment]
`0.11.4-a.1` works, thanks!

**Comment by Mr0grog** [comment]
Sorry it took me a while to publish a production release here. This is now available in v0.11.4.
